main.tf

```diff
@@ -50,19 +56,28 @@ locals {
           lookup(repo, "post_workflow_hooks", []),
           lookup(repo, "workflow", "") != "" && lookup(local._workflows, lookup(repo, "workflow", ""), "") != "" ? (
             local._workflows[lookup(repo, "workflow", "")].infracost.enabled ? [
-              { run : <<EOT
-if [ ! -d "/tmp/$BASE_REPO_OWNER-$BASE_REPO_NAME-$PULL_NUM" ]; then
-  exit 0
-fi
-
-infracost comment gitlab --repo $BASE_REPO_OWNER/$BASE_REPO_NAME \
-                         --merge-request $PULL_NUM \
-                         --path /tmp/$BASE_REPO_OWNER-$BASE_REPO_NAME-$PULL_NUM/'*'-infracost.json \
-                         --gitlab-token $GITLAB_TOKEN \
-                         --behavior new
-
-rm -rf /tmp/$BASE_REPO_OWNER-$BASE_REPO_NAME-$PULL_NUM
-EOT
+              { run : join("", [
+                "if [ ! -d \"/tmp/$BASE_REPO_OWNER-$BASE_REPO_NAME-$PULL_NUM\" ]; then exit 0; fi",
+                "\n\n",
+                join(" ", [
+                  "infracost",
+                  "comment",
+                  local._workflows[lookup(repo, "workflow", "")].infracost.platform,
+                  "--repo $BASE_REPO_OWNER/$BASE_REPO_NAME",
+                  "--merge-request $PULL_NUM",
+                  "--path /tmp/$BASE_REPO_OWNER-$BASE_REPO_NAME-$PULL_NUM/'*'-infracost.json",
+                  format("--%s $%s",
+                    local.infracost_parameters[local._workflows[repo.workflow].infracost.platform].token_name,
+                    coalesce(
+                      local._workflows[repo.workflow].infracost.token_environment_variable,
+                      local.infracost_parameters[local._workflows[repo.workflow].infracost.platform].environment_variable_name)
+                  ),
+                  "--behavior new",
```

Comment:
can we add the `behaviour` to the variable object?

Reply:
done